dipy/denoise/nlmeans.py

```diff
@@ -30,15 +30,27 @@ def nlmeans(arr, sigma, mask=None, patch_radius=1, block_radius=5,
     -------
     denoised_arr : ndarray
         the denoised ``arr`` which has the same shape as ``arr``.
+
+    References
+    ----------
+    .. [Descoteaux08] Descoteaux, Maxim and Wiest-Daessle`, Nicolas and Prima,
+                      Sylvain and Barillot, Christian and Deriche, Rachid
+                      Impact of Rician Adapted Non-Local Means Filtering on
+                      HARDI, MICCAI 2008
+
     """
 
     if arr.ndim == 3:
+        print("WARNING: This version of nlmeans has been deprecated")
```

Comment:
You need to use the warning system from Python. Here is an example from a different part of DIPY.
```
from warnings import warn
warn(DeprecationWarning("Module 'dipy.segment.quickbundles' is deprecated,"
			" use module 'dipy.segment.clustering' instead"))
```